Quickref.md

```diff
@@ -0,0 +1,221 @@
+-- list, current scroll position, attributes, and row content.  Returns current scroll position
+-- and current selection.  Full documentation in Reflex/Dom/Widget/Lazy.hs
+[W]   virtualListWithSelection :: Ord k => Int -> Int -> Dynamic Int -> Int -> Event Int ->
+          String -> Dynamic (Map String String) -> String -> Dynamic (Map String String) ->
+          (k -> Dynamic v -> m ()) -> Dynamic (Map k v) -> m (Dynamic (Int, Int), Event k)
+```
+
+## Connecting to the real world (I/O)
+
+### Connecting to DOM events
+
+```haskell
+-- Extract the specified Event from an 'El'.  The allowable event names, and their
+-- corresponding data types, are listed in Reflex/Dom/Widget/Basic.hs
+[ ]   domEvent :: EventName en -> El -> Event (EventResultType en)
+```
+
+### Performing arbitrary I/O in response to Events
+
+```haskell
+-- Run side-effecting actions in Event when it occurs; returned Event contains
+-- results. Side effects run in (WidgetHost m) monad, which includes [S] and [H]
+-- and can also do I/O via liftIO
+[W]   performEvent      :: Event (                WidgetHost m  a) -> m (Event a)
+
+-- Just run side-effects; no return Event
+[W]   performEvent_     :: Event (                WidgetHost m ()) -> m ()
+
+-- Actions run asynchronously; actions are given a callback to send return values
+[W]   performEventAsync :: Event ((a -> IO ()) -> WidgetHost m ()) -> m (Event a)
+```
+
+### XMLHttpRequest
+
+Convenience functions for XMLHttpRequest.  For configuration field details, see Reflex/Dom/Xhr.hs.
+
+```haskell
+-- Given method, URL, and config record (with default instance), construct a request.
+[ ]   xhrRequest :: String -> String -> XhrRequestConfig -> XhrRequest
+
+-- Given Event of requests, issue them and produce Event of responses.
+[W]   performRequestAsync :: Event XhrRequest -> m (Event XhrResponse)
+
+-- Issue a collection of requests, wait for them ALL to complete, return collected results.
+[W]   performRequestsAsync :: Traversable f => Event (f XhrRequest) -> m (Event (f XhrResponse))
+
+-- Convenience function to decode JSON-encoded response.
+[ ]   decodeXhrResponse :: FromJSON a => XhrResponse -> Maybe a
+
+-- Simplified interface to "GET" URLs and return decoded results.
+[W]   getAndDecode :: FromJSON a => Event String -> m (Event (Maybe a))
+```
+
+### Time
+
+```haskell
+-- Given a time interval in seconds and a basis time, create an Event that fires on the given
+-- interval. The Event will report the number of time intervals elapsed since the basis time
+-- (and other info, see Reflex/Dom/Time.hs).  High load will cause ticks to be dropped.
+[W]   tickLossy :: NominalDiffTime -> UTCTime -> m (Event t TickInfo)
+
+-- Delay an Event's occurences by a given amount in seconds.
+[W]   delay :: NominalDiffTime -> Event t a -> m (Event t a)
+```
+
+## Startup
+
+```haskell
+-- Reflex-Dom entry point.  Takes a monadic widget-building action of lengthy
+-- type and turns it into an IO action.
+[I]   mainWidget ::
```

Comment:
Maybe here is where you mention the MonadWidget signature.  It still might be good to show the Spider signatures here since these are where the type class finally gets made into the concrete Spider types.  Maybe you could put them both?